terraform/modules/gro_ingestion_service/lambdas.tf

```diff
@@ -0,0 +1,69 @@
+locals {
+  lambda_runtime = "nodejs18.x"
+  functions = toset([
+    for file in fileset("${path.module}/lambdas/src/functions", "*.function.ts") : lower(replace(split(".", split("/", file)[length(split("/", file)) - 1])[0], "[^a-zA-Z0-9]", "-"))
+  ])
+}
+
+data "archive_file" "typescript-source" {
+  type        = "zip"
+  source_dir  = "${path.module}/lambdas/src"
+  output_path = "zip/typescript-source.zip"
```

Comment:
So this is a way to get a hash of the whole directory, without hashing by hand? Do we _need_ to, or can we basically just always compile it and then just diff at the end? (Not sure how the complexity here vs the build time trade off though)

Reply:
Why diff after the compile rather than diff the source? diffing the source would just be quicker?

Reply:
It's more complex, and I  think my question is: is compiling it that much more expensive than zipping? For something small, I suspect it'll compile very quickly (but that's a naive statement)

Reply:
Compiling also involves running npm install, so I think it would be a fair bit longer imo, the zipping is incredibly quick